gateway/ln-gateway/src/auth_manager.rs

```diff
@@ -0,0 +1,59 @@
+use std::time::UNIX_EPOCH;
+
+use fedimint_core::secp256k1::PublicKey;
+use jsonwebtoken::{encode, EncodingKey, Header};
+use serde::{Deserialize, Serialize};
+
+const SESSION_EXPIRY_SECONDS: u64 = 60 * 30; // 30 minute
+
+pub struct AuthManager {
+    ///gateway id
+    gateway_id: PublicKey,
+    /// A secret key to encode a JWT with
+    pub encoding_secret: [u8; 16],
+}
+
+impl AuthManager {
+    /// Create a new auth manager
+    pub fn new(encoding_secret: [u8; 16], gateway_id: PublicKey) -> Self {
+        Self {
+            gateway_id,
+            encoding_secret,
+        }
+    }
+    pub fn generate_session(&self) -> anyhow::Result<Session> {
+        let session = Session::new(self.gateway_id, SESSION_EXPIRY_SECONDS);
+        Ok(session)
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Session {
+    /// the unique identifier of the session.
+    pub id: PublicKey,
+    /// the expire time of the session
+    pub exp: u64,
+}
```

Comment:
Member fields can be private afaik. Only the auth manager should access them to either create a session or check it.